### PR TITLE
[SPARK-42835][SQL][TESTS] Add test cases for `Column.explain`

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
Recently, I found Column.explain missing test cases.
This PR want add these test cases for easy to find the change if the `def toString` or `def sql` changed.

### Why are the changes needed?
Add test cases for Column.explain

### Does this PR introduce _any_ user-facing change?
'No'.
Just add test cases for Column.explain.

### How was this patch tested?
New test cases.

### CODE REVIEW

**Comment by amaliujia** [comment]
Just a general question which is for my self education: do we expect the results of `Column.explain` are stable?

**Comment by beliefer** [comment]
> Just a general question which is for my self education: do we expect the results of `Column.explain` are stable?

Personally, I think Spark should keep the stable output and do not break change for users.

File: sql/core/src/test/scala/org/apache/spark/sql/ColumnExpressionSuite.scala [hunk 1]
```diff
@@ -921,6 +922,132 @@ class ColumnExpressionSuite extends QueryTest with SharedSparkSession {
     }
   }
 
+  private def captureStdOut(block: => Unit): String = {
+    val capturedOut = new ByteArrayOutputStream()
+    Console.withOut(capturedOut)(block)
+    capturedOut.toString()
+  }
+
+  test("explain") {
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/ColumnExpressionSuite.scala:931]: Hm, I wouldn't add a bunch of tests though. `expr.sql` isn't actually strictly providing the compatibility in its string representation, and this test case potentially fails often.

Comment (amaliujia) [reply sql/core/src/test/scala/org/apache/spark/sql/ColumnExpressionSuite.scala:931]: yeah that was what I was wondering if this output is expected stable enough. If not then probably we should not add such tests.

Comment (beliefer) [reply sql/core/src/test/scala/org/apache/spark/sql/ColumnExpressionSuite.scala:931]: > Hm, I wouldn't add a bunch of tests though. `expr.sql` isn't actually strictly providing the compatibility in its string representation, and this test case potentially fails often.

In fact, explain not only reactor `expr.sql`, and `expr.toString` too.

**Comment by beliefer** [comment]
ping @cloud-fan

File: sql/core/src/test/scala/org/apache/spark/sql/ColumnExpressionSuite.scala (hunk 1 above)
Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/ColumnExpressionSuite.scala:931]: shall we use golden file tests? We don't expect it to be stable but we should be aware if it gets changed.

Comment (amaliujia) [reply sql/core/src/test/scala/org/apache/spark/sql/ColumnExpressionSuite.scala:931]: +1 if we still want be aware it's changing, golden files is a better way.

**Comment by cloud-fan** [comment]
Can we use the existing golden file framework? I think we just need to add a bunch of test queries like `SELECT a + b`.

**Comment by beliefer** [comment]
> EDIT: sorry I just realized that this PR is to test the result of `Column.explain`. Why do we care about its stability? The API doc says `Prints the expression to the console for debugging purposes`. We don't have stability tests for the plan EXPLAIN result either.

Got it. I will close this PR.

**Pull request closed without merging**